tests/src/test/java/org/eclipse/hono/tests/amqp/AmqpUploadTestBase.java

```diff
@@ -241,6 +244,76 @@ public void testAdapterClosesLinkOnMessageExceedingMaxPayloadSize(final VertxTes
+     * authority.
+     *
+     * @param senderQos The delivery semantics to use for the device.
+     * @throws InterruptedException if test is interrupted while running.
+     */
+    @ParameterizedTest(name = IntegrationTestSupport.PARAMETERIZED_TEST_NAME_PATTERN)
+    @MethodSource("senderQoSTypes")
+    public void testAutoProvisioningViaGateway(final ProtonQoS senderQos) throws InterruptedException {
+
+        final String tenantId = helper.getRandomTenantId();
+        final String gatewayId = helper.getRandomDeviceId(tenantId);
+        final Device gateway = new Device()
+                .setAuthorities(Collections.singleton(RegistryManagementConstants.AUTHORITY_AUTO_PROVISIONING_ENABLED));
+
+        final String username = IntegrationTestSupport.getUsername(gatewayId, tenantId);
+        final VertxTestContext emptyEventReceived = new VertxTestContext();
+        helper.applicationClientFactory.createEventConsumer(tenantId,
+                msg -> emptyEventReceived.verify( () -> {
+                    assertThat(msg.getContentType()).isEqualTo(EventConstants.CONTENT_TYPE_EMPTY_NOTIFICATION);
+                    assertThat(MessageHelper.getRegistrationStatus(msg)).isEqualTo(EventConstants.RegistrationStatus.NEW.name());
+                    emptyEventReceived.completeNow();
+                }),
+                close -> {});
+
+        final VertxTestContext telemetryReceived = new VertxTestContext();
+        createConsumer(tenantId, msg -> {
+            telemetryReceived.verify( () -> {
+                assertThat(msg.getContentType()).isEqualTo("text/plain");
+                assertMessageProperties(telemetryReceived, msg);
+            });
+            telemetryReceived.completeNow();
+        });
+
+        final VertxTestContext setup = new VertxTestContext();
+        helper.registry
+                .addDeviceForTenant(tenantId, new Tenant(), gatewayId, gateway, DEVICE_PASSWORD)
+                .compose(ok -> connectToAdapter(username, DEVICE_PASSWORD))
+                .compose(con -> createProducer(null, senderQos))
+                .recover(t -> {
+                    log.error("error setting up AMQP protocol adapter", t);
+                    return Future.failedFuture(t);
+                })
+                .onComplete(setup.succeeding(s -> {
+                    sender = s;
+                    setup.completeNow();
+                }));
+
+        assertThat(setup.awaitCompletion(5, TimeUnit.SECONDS)).isTrue();
+        assertThat(setup.failed())
+                .as("successfully connect to adapter")
+                .isFalse();
+
+        final VertxTestContext messageSent = new VertxTestContext();
+        final Message msg = ProtonHelper.message("apFoobar");
+        msg.setContentType("text/plain");
+
+        final String deviceId = helper.getRandomDeviceId(tenantId);
+        msg.setAddress(String.format("%s/%s/%s", getEndpointName(), tenantId, deviceId));
+        sender.send(msg, delivery -> {
```

Comment:
you should either consider the senderQoS being passed into the test method or do not run this as a parameterized test. FMPOV the latter would be ok.